src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/fsm/path/HistoryFormatter.java

```diff
@@ -0,0 +1,46 @@
+/* Copyright 2022 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flowhs.fsm.path;
+
+import org.openkilda.floodlight.api.request.FlowSegmentRequest;
+import org.openkilda.floodlight.api.response.SpeakerFlowSegmentResponse;
+import org.openkilda.floodlight.flow.response.FlowErrorResponse;
+import org.openkilda.wfm.topology.flowhs.model.path.FlowPathRequest;
+import org.openkilda.wfm.topology.flowhs.model.path.FlowPathRequest.PathChunkType;
+import org.openkilda.wfm.topology.flowhs.service.common.FlowHistoryCarrier;
+
+import java.util.List;
+
+abstract class HistoryFormatter {
```

Comment:
Like the idea. But can the name be more specific? Once we reuse this concept we'll have another more generic HistoryFormatter...

Reply:
Ok. Renamed it into `FlowPathHistoryFormatter`.